### DISCUSSION TITLE
How curl switches from GET to POST and to GET back during a series of 302 redirect

### DISCUSSION THREAD

**Discussion by eppesuig** [curl/curl#13121] [Q&A] [answered]
Hello,
I am facing a problem when using curl for accessing a web service that requires an authentication similar to SAML, and that exchanges authentication data between many hosts using http status code 302.

The problem is that some of these calls require to being called via method POST, and some other via GET.

How does curl select the method to use? How may I specify one of such method for the 2nd or 2rd call?

I tried using curl on the command line, starting with a POST method, and noticed that it correctly commute from POST to GET on the next calls. But, when I use a commercial library that internally uses libcurl, I seel all calls use POST and get eventually rejected.

Is there any parameters that the library might have used in order to force this behaviour?

Thank you,
Giuseppe S.

**Comment by dfandrich** [comment, marked as answer]
That sounds like a difference that could be explained by [CURLOPT_POSTREDIR](https://curl.se/libcurl/c/CURLOPT_POSTREDIR.html). curl doesn't offer an option to switch based on the number of the redirect, but rather the specific redirect code. If you want to do the former, then the code will need to handle redirects itself.

**Comment by eppesuig** [comment]
Thank you, @dfandrich , I tried to set that option to 0, but it made no difference. I suppose the problem is in the library itself that might set it back somehow. I'll contact the vendor in order to check it.
